presto-orc/src/test/java/com/facebook/presto/orc/TestOrcBloomFilters.java

```diff
@@ -0,0 +1,333 @@
+            }
+            else if (o instanceof String) {
+                bloomFilter.addString((String) o);
+            }
+            else if (o instanceof BigDecimal) {
+                bloomFilter.addString(o.toString());
+            }
+            else if (o instanceof Slice) {
+                bloomFilter.addString(((Slice) o).toStringUtf8());
+            }
+            else if (o instanceof Timestamp) {
+                bloomFilter.addLong(((Timestamp) o).getTime());
+            }
+            else if (o instanceof Double) {
+                bloomFilter.addDouble((Double) o);
+            }
+            else {
+                fail("Unsuppored type " + o.getClass());
+            }
+        }
+
+        for (Map.Entry<Object, Type> testValue : writeValues.entrySet()) {
+            boolean matched = TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, testValue.getKey(), testValue.getValue());
+            assertTrue(matched, "type " + testValue.getClass());
+        }
+
+        // test unsupported type
+        assertTrue(TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, new Date(), TIMESTAMP), "unsupported type Date should always return true");
+    }
+
+    public ImmutableMap<Object, Type> getTestElements()
+    {
+        ImmutableMap.Builder<Object, Type> writeBuilder = ImmutableMap.builder();
+        writeBuilder.put(TEST_STRING, VARCHAR);
+        writeBuilder.put(4312L, BIGINT);
+        writeBuilder.put(123, INTEGER);
+        writeBuilder.put(234D, DOUBLE);
+        return writeBuilder.build();
+    }
+
+    @Test
+    public void testBloomFilterPredicateValuesNonExisting()
+            throws Exception
+    {
+        ImmutableMap<Object, Type> testValues = getTestElements();
+
+        BloomFilter bloomFilter = new BloomFilter(testValues.size() * 10, 0.01D);
+
+        for (Map.Entry<Object, Type> testValue : testValues.entrySet()) {
+            boolean matched = TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, testValue.getKey(), testValue.getValue());
+            assertFalse(matched, "type " + testValue.getKey().getClass());
+        }
+
+        // test unsupported type
+        assertTrue(TupleDomainOrcPredicate.checkInBloomFilter(bloomFilter, new Date(), null), "unsupported type Date should always return true");
+    }
+
+    @Test
+    public void testExtractValuesFromSingleDomain()
+            throws Exception
+    {
+        ImmutableMap.Builder<Type, Object> testBuilder = ImmutableMap.builder();
+        testBuilder.put(BOOLEAN, true);
+        testBuilder.put(INTEGER, 1234L); // internally treated as long
+        testBuilder.put(BIGINT, 4321L);
+        testBuilder.put(DOUBLE, 0.123D);
+        BlockBuilder blockBuilder = VARCHAR.createBlockBuilder(new BlockBuilderStatus(), 15);
+        VARCHAR.writeString(blockBuilder, TEST_STRING);
+        Block block = blockBuilder.build();
+        testBuilder.put(VARCHAR, VARCHAR.getSlice(block, 0));
```

Comment:
This can be `testBuilder.put(VARCHAR, Slices.utf8Slice(TEST_STRING))`, but static import utf8Slice